### PR TITLE
Fix grid template parsing in case of auto row sizes

### PR DESCRIPTION
**Pull request by y-gagar1n**
#1049 I added some additional checks so that row size would be defined as 'auto' if not specified.

### CODE REVIEW

**Comment by yepninja** [comment]
@y-gagar1n This case is about `grid-gap`, so it's better to put tests into `grid-gap.css`.

Also, you can use simpler logic. After `grid-template` parsed you can check rows. If there are no rows, you can add `auto` values according to areas size.

**Comment by yepninja** [comment]
Now I understand your test cases. So we need at least two tests in different files (without and with gap).

File: lib/hacks/grid-utils.js [hunk 1]
```diff
@@ -200,6 +200,14 @@ function testTrack(node) {
     return node.type === 'word' && /^\[.+\]$/.test(node.value);
 }
 
+function verifyRowSize(result) {
+    if (result.prop === 'grid-template' &&
```
Comment (yepninja) [inline lib/hacks/grid-utils.js]: It doesn't need to check prop value. It's always `grid-template`.

File: lib/hacks/grid-utils.js [hunk 2]
```diff
@@ -213,22 +221,34 @@ function parseTemplate({
...
+                    result.sizedRow = true;
+                }
             }
 
             // devider(/)
             if (type === 'div' && value === '/') {
                 result.key = 'columns';
+                result = verifyRowSize(result);
+                result.rowProps = false;
             }
```
Comment (yepninja) [inline lib/hacks/grid-utils.js]: Maybe it is enough to compare the length of areas and rows. And if the length of rows is less then the length of areas it needed to push `auto` to rows array.

**Comment by yepninja** [comment]
Maybe it is enough to compare the length of areas and rows. And if the length of rows is less then the length of areas it needed to push auto to rows array.

**Comment by ai** [comment]
@yepninja we are ready to merge?

**Comment by y-gagar1n** [comment]
@ai @yepninja sorry, I'm not finished yet. Commit placing tests in different files will be ready a bit later.

**Comment by y-gagar1n** [comment]
> Maybe it is enough to compare the length of areas and rows. And if the length of rows is less then the length of areas it needed to push auto to rows array.

I still don't think it's possible to handle the case with mixed sized and unsized rows that way.  Imagine we had two different cases:

```css
.a {
  grid-template:
      "a b"
      "c d" 20px;
}

.b {
  grid-template:
      "a b" 20px
      "c d" ;
}
```

Both of them would output `rows` equal to `20px` and information about which row had this size would be lost completely during the `parseTemplate` function. Appending `auto` will result in `20px auto` which is wrong in case of `.a`.

**Comment by y-gagar1n** [comment]
@ai @yepninja I'm finished now.

File: lib/hacks/grid-utils.js [hunk 3]
```diff
@@ -200,6 +200,13 @@ function testTrack(node) {
     return node.type === 'word' && /^\[.+\]$/.test(node.value);
 }
 
+function verifyRowSize(result) {
+    if (result.rowProps && !result.sizedRow) {
```
Comment (yepninja) [inline lib/hacks/grid-utils.js]: I mean that you can just compare length here.
```js
if (result.areas.length > result.rows.length) result.rows.push('auto')
```
So you don't need additional properties (`rowProps` and `sizedRowz`)

Comment (y-gagar1n) [reply lib/hacks/grid-utils.js]: Thanks, now I finally see what you mean. I'll commit later today.

**Comment by yepninja** [comment]
@y-gagar1n Great PR! Thank you for contributing!
@ai Now we are ready to merge.

**Comment by ai** [comment]
Thanks. I will release it tomorrow.

**Comment by ai** [comment]
Released in 8.5.2.

**Pull request merged**